slick/slick.js

```diff
@@ -2933,6 +2942,41 @@ Issues: http://github.com/kenwheeler/slick/issues
 		return out;
 	};
 
+	// @param  {Array}|{NodeList} `collectionOfNode` A collection of node to filter
+	// @param  {function}|{cssSelector}|{Node} `filter` The filter param
+	// @return {Array} of HTMLElement filtred by `filter`
+	// @usage Slick.filterNodeUtil(collectionOfNode, (funtion(elem,index,array){...} | '.my-selector' | node) );
+	// Equivalent to jQuery.filter() method, BUT do not support extended jQuery selector https://api.jquery.com/category/selectors/jquery-selector-extensions/
+	Slick.prototype.filterNodeUtil = function(collectionOfNode, filter) {
+		var _ = this,
+			domParent = document.createElement("div"),
+			selectedNodes,
+			filterFunction;
+
+		if (typeof filter === "string" && filter) {
+			Array.prototype.forEach.call(collectionOfNode, function(elem) {
+				domParent.appendChild(elem);
+			});
+			selectedNodes = _.queryAll(filter, domParent);
+			filterFunction = function(elem) {
+				return selectedNodes.lastIndexOf(elem) >= 0;
+			};
```

Comment:
`var _ = this,			
			selectedNodes = [],
			filterFunction;

		if (typeof filter === "string" && filter) {
			Array.prototype.forEach.call(collectionOfNode, function(elem) {				
                if(elem.matches(filter)) {                  
                    selectedNodes.push(elem);
                }               
			});
          
			filterFunction = function(elem) {
				return selectedNodes.lastIndexOf(elem) >= 0;
			};`